contracts/VotingMachines/Governance.sol

```diff
@@ -0,0 +1,509 @@
+pragma solidity ^0.4.18;
+
+import "../controller/Reputation.sol";
+import "./IntVoteInterface.sol";
+import "../universalSchemes/UniversalScheme.sol";
+import "./GovernanceFormulasInterface.sol";
+
+
+
+contract Governance is IntVoteInterface,UniversalScheme,GovernanceFormulasInterface {
+    using SafeMath for uint;
+
+    enum ProposalState { Closed, Executed, NoneBoosted, Boosted }
+
+    struct Parameters {
+        Reputation reputationSystem; // the reputation system that is being used
+        uint noneBoostedVoteRequierePercentage; // the absolute vote percentages bar.
+        uint noneBoostedVotePeriodLimit; //the time limit for a proposal to be in an absolute voting mode.
+        uint boostedVotePeriodLimit; //the time limit for a proposal to be in an relative voting mode.
+        uint scoreThreshold;
+        GovernanceFormulasInterface governanceFormulasInterface;
+        uint minimumStaketingFee;
+    }
+    struct Voter {
+        uint vote; // 0 - 'abstain'
+        uint reputation; // amount of voter's reputation
+    }
+
+    struct Staker {
+        uint vote; // 0 - 'abstain'
+        uint amount; // amount of voter's reputation
+    }
+
+    struct Proposal {
+        address avatar; // the organization's avatar the proposal is target to.
+        uint numOfChoices;
+        ExecutableInterface executable; // will be executed if the proposal will pass
+        uint totalVotes;
+        uint totalStakes;
+        uint submittedTime;
+        uint boostedPhaseTime; //the time the proposal shift to relative mode.
+        ProposalState state;
+        uint winningVote; //the winning vote.
+        mapping(uint=>uint) votes;
+        mapping(address=>Voter) voters;
+        mapping(uint=>uint) stakes;
+        mapping(address=>Staker) stakers;
+    }
+
+    event NewProposal(bytes32 indexed _proposalId, uint _numOfChoices, address _proposer, bytes32 _paramsHash);
```

Comment:
These events have a different naming convention than `AbsoluteVote`.  I think they should be consistent.  AbsoluteVote events (and I think those in other contracts as well) have the word "Log" prepended, like this:

```
event LogNewProposal(bytes32 indexed _proposalId, uint _numOfChoices, address _proposer, bytes32 _paramsHash);
    event LogCancelProposal(bytes32 indexed _proposalId);
    event LogExecuteProposal(bytes32 indexed _proposalId, uint _decision);
    event LogVoteProposal(bytes32 indexed _proposalId, address indexed _voter, uint _vote, uint _reputation, bool _isOwnerVote);
    event LogCancelVoting(bytes32 indexed _proposalId, address indexed _voter);
```

Reply:
Will change AbsoluteVote  events as we do not need the "Log" .It is obvious that it is an event.

Reply:
Update at a6ba131

Reply:
I think we should make this change everywhere.   So I created a separate issue for it:  

https://github.com/daostack/Arc/issues/282

Reply:
ok. not much left